deps/ReactantExtra/src/error_handling.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "absl/status/statusor.h"
+#include "xla/pjrt/exceptions.h"
+#include "xla/pjrt/status_casters.h"
+
+extern "C" extern void (*ReactantThrowError)(const char *);
```

Comment:
Locally I get
````
[05:42:43] ERROR: /workspace/srcdir/Reactant.jl/deps/ReactantExtra/BUILD:355:11: Compiling src/ifrt/DynamicShape.cpp failed: (Exit 1): cc failed: error executing command (from target //:ReactantExtraLib)
[...]
[05:42:43] In file included from src/ifrt/DynamicShape.cpp:2:
[05:42:43] ./src/error_handling.hpp:7:12: error: invalid use of ‘extern’ in linkage specification
[05:42:43]     7 | extern "C" extern void (*ReactantThrowError)(const char *);
[05:42:43]       |            ^~~~~~
[05:42:44] [8,094 / 8,136] Compiling src/enzyme_ad/jax/TransformOps/TransformOps.cpp; 20s processwrapper-sandbox ... (41 actions running)
[05:42:45] [8,104 / 8,136] Compiling src/ifrt/Client.cpp; 12s processwrapper-sandbox ... (31 actions running)
[05:42:46] Target //:libReactantExtra.so failed to build
```
I presume this should be
```suggestion
extern "C" void (*ReactantThrowError)(const char *);
```
?

Reply:
i remember i wrote like that before but failed (on aarch64-macos). let me try again first locally.

Reply:
This is inconsistent with https://github.com/EnzymeAD/Reactant.jl/blob/e1e62cf654ec6a07666451d2819fd8de684c2d6c/deps/ReactantExtra/src/error_handling.cpp#L1

> also, the build succeeds on aarch64-macos

Not very useful to say "another compiler is happy with broken stuff", is it? :slightly_smiling_face:

Reply:
> and `extern` is used to instruct that the symbol is declared in another object file, so don't declare it.

But this is a header file and the implementation is in the `.cpp` file.  Why do you need to mark the symbol as `extern` in a header?

Reply:
look that `MyValueOrThrow` is a template function, so it must go on the header and it calls `ReactantThrowError`. all files that use `MyValueOrThrow` must know about it

Reply:
This isn't about `MyValueOrThrow` but `ReactantThrowError`.